hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/impl/TestContainerDataYaml.java

```diff
@@ -233,28 +236,48 @@ public void testChecksumInContainerFile() throws IOException {
     // Read from .container file, and verify data.
     KeyValueContainerData kvData = (KeyValueContainerData) ContainerDataYaml
         .readContainerFile(containerFile);
-    ContainerUtils.verifyChecksum(kvData);
+    ContainerUtils.verifyChecksum(kvData, conf);
 
     cleanup();
   }
 
+  private KeyValueContainerData getKeyValueContainerData() throws IOException {
+    String containerFile = "incorrect.checksum.container";
+    //Get file from resources folder
+    ClassLoader classLoader = getClass().getClassLoader();
+    File file = new File(classLoader.getResource(containerFile).getFile());
+    return (KeyValueContainerData) ContainerDataYaml.readContainerFile(file);
+  }
+
   /**
    * Test to verify incorrect checksum is detected.
    */
   @Test
   public void testIncorrectChecksum() {
     try {
-      String containerFile = "incorrect.checksum.container";
-      //Get file from resources folder
-      ClassLoader classLoader = getClass().getClassLoader();
-      File file = new File(classLoader.getResource(containerFile).getFile());
-      KeyValueContainerData kvData = (KeyValueContainerData) ContainerDataYaml
-          .readContainerFile(file);
-      ContainerUtils.verifyChecksum(kvData);
+      KeyValueContainerData kvData = getKeyValueContainerData();
+      ContainerUtils.verifyChecksum(kvData, conf);
       fail("testIncorrectChecksum failed");
     } catch (Exception ex) {
       GenericTestUtils.assertExceptionContains("Container checksum error for " +
           "ContainerID:", ex);
     }
   }
+
+  /**
+   * Test to verify disabled checksum with incorrect checksum.
+   */
+  @Test
+  public void testDisabledChecksum(){
+    try {
+      KeyValueContainerData kvData = getKeyValueContainerData();
+      conf.setBoolean(HddsConfigKeys.
+                      HDDS_CONTAINER_CHECKSUM_VERIFICATION_ENABLED, false);
+      ContainerUtils.verifyChecksum(kvData, conf);
+    } catch (Exception ex) {
+      ex.printStackTrace();
```

Comment:
nit. can we remove this printStackTrace?

Reply:
Sure. I can remove it.